homeassistant/components/conversation.py

```diff
@@ -29,20 +31,43 @@
     vol.Required(ATTR_TEXT): vol.All(cv.string, vol.Lower),
 })
 
-CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({}),
-}, extra=vol.ALLOW_EXTRA)
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({
+    cv.string: vol.Schema({
+        vol.Required('keywords'): cv.string,
+        vol.Required('action'): cv.SCRIPT_SCHEMA,
+    })
+})}, extra=vol.ALLOW_EXTRA)
 
 
 def setup(hass, config):
     """Register the process service."""
-    warnings.filterwarnings('ignore', module='fuzzywuzzy')
     from fuzzywuzzy import process as fuzzyExtract
+    warnings.filterwarnings('ignore', module='fuzzywuzzy')
 
     logger = logging.getLogger(__name__)
+    config = config.get(DOMAIN, {})
+    # Right way to do this-- taken from updater.py
+
+    choices = {attrs['keywords']: script.Script(
+        hass,
+        attrs['action'],
+        name)
+        for name, attrs in config.items()} or {'': lambda: None}
```

Comment:
It's fine for choices to be empty. In that case we should just keep the original behavior of turn_on and turn_off without any extra sentences.